ui-core/src/components/inputs/Slider.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import type { ChangeEvent, MouseEvent, InputHTMLAttributes } from 'react';
+
+import cx from 'classnames';
+
+export type SliderProps = InputHTMLAttributes<HTMLInputElement> & {
+  onChangeCommitted?: (e: MouseEvent<HTMLInputElement>) => void;
+};
+
+// onChange returns an event or number
+const Slider = ({
+  id,
+  value: initialValue,
+  min = 0,
+  max = 100,
+  step = 1,
+  onChange,
+  onChangeCommitted,
+  disabled,
+  ...rest
+}: SliderProps) => {
+  const [value, setValue] = useState<number>(
+    initialValue !== undefined ? Number(initialValue) : Number(min)
+  );
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newValue = Number(e.target.value);
+    setValue(newValue);
+    onChange?.(e);
+  };
+
+  const handleMouseUp = (e: MouseEvent<HTMLInputElement>) => {
+    if (onChangeCommitted) {
+      onChangeCommitted(e);
+    }
+  };
+
+  return (
+    <div className="range-wrapper">
+      <input
+        type="range"
+        className={cx('range-slider', { 'range-slider-disabled': disabled })}
+        id={id}
+        value={value}
+        min={min}
+        max={max}
+        step={step}
+        onChange={handleChange}
+        onMouseUp={handleMouseUp}
```

Comment:
In what case would we want a different behavior than the `onChange` ?

Reply:
onChange triggers on every single frame. There are definitely some case we don’t we to trigger expensive computation or rerender on every frame.